tests/expressions/test_parser.py

```diff
@@ -199,3 +200,8 @@ def test_with_function() -> None:
         parser.parse("foo = 1 and lower(bar) = '2'")
 
     assert "Expected end of text, found 'and'" in str(exc_info)
+
+
+def test_cast() -> None:
+    cast = parser.parse("CAST(created_at as date)")
```

Comment:
I would love to see some more tests here. The most important part that's missing here is converting `date` to a `DayTransform`. Other options such as `year`, `month` etc should be added as well.